fastMatcher.go

```diff
@@ -548,7 +577,7 @@ func (m *FastMatcher) matchExec(token tokenType, tokenData []byte, tokenDataLen
 			}
 		}
 	} else if token == tknObjectStart {
-		objStartPos := m.tokens.Position()
+		objStartPos := m.tokens.Position() - 1 /* to include the objStart itself*/
```

Comment:
This does not seem related to the slots changes.  What is this?

Reply:
Just something that when I was doing testing I found the obj FastVal was w/o the "{"
Since this PR also addresses the issue of doing object comparison, I took the liberty and fixed that here.